cmake/mission_build.cmake

```diff
@@ -125,9 +125,16 @@ function(prepare)
   # Locate the source location for all the apps found within the target file
   # Each of those may in turn have a "mission_build" file that calls out additional dependencies for that app,
   # so this is run in a loop until the list of unfound apps is empty
-  string(REPLACE ":" ";" CFS_APP_PATH "$ENV{CFS_APP_PATH}")
-  list(APPEND CFS_APP_PATH "apps" "apps/CFS" "libs" "psp/fsw/modules")
-  set(MISSION_DEPS "cfe-core" "osal")
+  string(REPLACE ":" ";" CFS_USER_APP_PATH "$ENV{CFS_APP_PATH}")
+  set(CFS_APP_PATH 
+    ${CFS_USER_APP_PATH}    # User-supplied/mission-specific paths are checked first
+    "apps"                  # general purpose $[top}/apps directory
+    "libs"                  # general purpose $[top}/libs directory
+    "psp/fsw/modules"       # modules for optional platform abstraction, associated with PSP
+    "cfe/modules"           # modules for optional core functions, associated with CFE
+     CACHE STRING "Search path for code modules"
+  )
+  set(MISSION_DEPS "cfe-core" "osal" ${MISSION_CORE_MODULES})
```

Comment:
Suggest adding the hard coded directories to the "mission defaults".  And add a status message listing the search path.

Reply:
So you are suggesting not having any standard directories out of the box?
Adding a message to show the complete search path is fine, that's easy enough.

Reply:
Nope, I think we should simplify the "magic" (hard coded include paths, hard coded search paths, hard coded target linkage, etc) scattered about CMake and define things like this in the mission_default.cmake.  Then the logic just works on the set variables, rather than defining the variables.

And/or where it makes sense transition to letting targets define the includes.  Looks like there's a start with the osal target properties (includes and defines), but I'd think if they all did it with interface targets less magic would be needed at the top level.  Just inherit properties up to the top level, then set up the apps using the interface properties of the dependent targets.  Couple that with a simple function apps could call to add the ut_assert lib interface from their CMakeList or whatever.